stdlib/strings/strings.go

```diff
@@ -466,8 +466,11 @@ func init() {
 				if !ok {
 					return nil, fmt.Errorf("missing argument %q", "arr")
 				}
-				if val.Type().Nature() != semantic.Array {
-					return nil, fmt.Errorf("expected argument %q to be of type %v, got type %v", "arr", semantic.Array, val.Type().Nature())
+				arr := val.Array()
+				if arr.Len() > 0 {
+					if arr.Get(0).Type().Nature() != semantic.String {
```

Comment:
I don't think this will work right if `arr` has zero elements.  You can still peek at the type though, maybe try this?
```suggestion
					if arr.Type().ElementType().Nature() != semantic.String {
```